cpp/src/arrow/acero/hash_join_benchmark.cc

```diff
@@ -104,7 +104,7 @@ class JoinBenchmark {
       key_cmp.push_back(JoinKeyCmp::EQ);
     }
 
-    for (size_t i = 0; i < settings.build_payload_types.size(); i++) {
+    for (size_t i = 0; i < settings.probe_payload_types.size(); i++) {
```

Comment:
Does this change the benchmark results somehow?

Reply:
It is supposed to. But I didn't review the complete benchmarks that are utilizing this piece of code. There might be a chance that all benchmarks are using the same number of payload columns thus no result change. 

Is there anything we should be preparing for a potential benchmark result change?

Reply:
Well, it seems the "detailed" benchmarks all have:
```c++
  settings.build_payload_types = {binary()};
  settings.probe_payload_types = {binary()};
```
while the "basic" benchmarks leave those settings empty? (is it desired?)

Reply:
Ah, it would actually matter for `BM_HashJoinBasic_PayloadSize`.

Reply:
Ok, it seems to require OpenMP, so I won't try running it.